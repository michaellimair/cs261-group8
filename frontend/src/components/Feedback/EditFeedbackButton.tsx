import { Button } from '@chakra-ui/react';
import { FC } from 'react';
import { useTranslation } from 'react-i18next';

const EditFeedbackButton: FC<{ isLoading: boolean }> = ({
  isLoading,
}) => {
  const { t } = useTranslation();

  return (
    <Button
      data-testid="edit-feedback-button"
      colorScheme="green"
      isLoading={isLoading}
      type="submit"
      loadingText={t('dashboard.edit_feedback.loading')}
    >
      {t('dashboard.edit_feedback.title')}
    </Button>
  );
};

export default EditFeedbackButton;
