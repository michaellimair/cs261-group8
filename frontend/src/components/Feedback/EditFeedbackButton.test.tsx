import { render } from '@testing-library/react';
import EditFeedbackButton from './EditFeedbackButton';

describe('components/Feedback/EditFeedbackButton', () => {
  it('renders correctly', () => {
    const result = render(<EditFeedbackButton isLoading />);

    expect(result).toMatchSnapshot();
  });
});
