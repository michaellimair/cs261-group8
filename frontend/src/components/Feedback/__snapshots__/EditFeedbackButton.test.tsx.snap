// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Feedback/EditFeedbackButton renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <button
        class="chakra-button css-5c2176"
        data-loading=""
        data-testid="edit-feedback-button"
        disabled=""
        type="submit"
      >
        <div
          class="chakra-button__spinner chakra-button__spinner--start css-i5qkey"
        >
          <div
            class="chakra-spinner css-1plzyg6"
          >
            <span
              class="css-f8n5zr"
            >
              Loading...
            </span>
          </div>
        </div>
        dashboard.edit_feedback.loading
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="chakra-button css-5c2176"
      data-loading=""
      data-testid="edit-feedback-button"
      disabled=""
      type="submit"
    >
      <div
        class="chakra-button__spinner chakra-button__spinner--start css-i5qkey"
      >
        <div
          class="chakra-spinner css-1plzyg6"
        >
          <span
            class="css-f8n5zr"
          >
            Loading...
          </span>
        </div>
      </div>
      dashboard.edit_feedback.loading
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
